src/components/modals/ConnectWalletModal/ConnectWalletModal.tsx

```diff
@@ -110,6 +131,59 @@ function ConnectWalletModal(): JSX.Element {
     }
   }
 
+  useEffect(() => {
+    const handleEthAccount = async () => {
+      if (!ethAccount?.address) {
+        return;
+      }
+      if (!viemConnector) {
+        throw new Error('Unable to find Viem connector for Metamask');
+      }
+
+      try {
+        const connector = new EthWalletConnector(
+          ethAccount,
+          ethAccount.address,
+          ethDisconnect,
+          signMessage,
+          viemConnector,
+        );
+
+        const arweaveGate = await connector.getGatewayConfig();
+        const contract = IO.init({
+          process: new AOProcess({
+            processId: ioProcessId,
+            ao: aoClient,
+          }),
+          signer: connector.contractSigner!,
+        });
+        if (arweaveGate?.host) {
+          await dispatchNewGateway(
+            arweaveGate.host,
+            contract,
+            dispatchGlobalState,
+          );
+        }
+
+        const address = await connector.getWalletAddress();
+        dispatchWalletState({
+          type: 'setWalletAddress',
+          payload: address,
+        });
+        dispatchWalletState({
+          type: 'setWallet',
+          payload: connector,
+        });
+
+        closeModal({ next: true, address });
+      } catch (error: any) {
+        eventEmitter.emit('error', error);
+      }
+    };
+
+    handleEthAccount();
+  }, [ethAccount]);
```

Comment:
recommend a reducer action util for this - dispatchArNSWalletConnector or some such.

Reply:
Think it may not be necessary now that I moved to wagmi config-based EthWalletConnector. (Managed to remove this block of code altogether.)